client/state/happychat/middleware.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Internal dependencies
+ */
+import { ROUTE_SET } from 'state/action-types';
+import { sendEventMessage } from 'state/happychat/actions';
+import {
+	isHappychatClientConnected,
+	isHappychatAcceptingChats,
+	isHappychatChatAssigned
+ } from 'state/happychat/selectors';
+import { getCurrentUser } from 'state/current-user/selectors';
+
+export default ( store ) => ( next ) => ( action ) => {
+	if ( action.type === ROUTE_SET ) {
```

Comment:
We should use a `switch` here since we'll be handling quite a few different action types in the middleware in the future.